packages/vaex-core/vaex/dataframe_protocol.py

```diff
@@ -0,0 +1,725 @@
+        The values of an expression can be NumPy or Arrow.
+        """
+        if not isinstance(column, vaex.expression.Expression):
+            raise NotImplementedError("Columns of type {} not handled "
+                                      "yet".format(type(column)))
+
+        # Store the column as a private attribute
+        self._col = column
+        self._allow_copy = allow_copy
+        
+    @property
+    def size(self) -> int:
+        """
+        Size of the column, in elements.
+        """
+        return self._col.df.count("*")
+    
+    @property
+    def offset(self) -> int:
+        """
+        Offset of first element. Always zero.
+        """
+        return 0
+        
+    @property
+    def dtype(self) -> Tuple[enum.IntEnum, int, str, str]:
+        """
+        Dtype description as a tuple ``(kind, bit-width, format string, endianness)``
+        
+        Kind :
+        
+            - INT = 0
+            - UINT = 1
+            - FLOAT = 2
+            - BOOL = 20
+            - STRING = 21   # UTF-8
+            - DATETIME = 22
+            - CATEGORICAL = 23
+            
+        Bit-width : the number of bits as an integer
+        Format string : data type description format string in Apache Arrow C
+                        Data Interface format.
+        Endianness : current only native endianness (``=``) is supported
+        
+        Notes:
+        
+            - Kind specifiers are aligned with DLPack where possible (hence the
+              jump to 20, leave enough room for future extension)
+            - Masks must be specified as boolean with either bit width 1 (for bit
+              masks) or 8 (for byte masks).
+            - Dtype width in bits was preferred over bytes
+            - Endianness isn't too useful, but included now in case in the future
+              we need to support non-native endianness
+            - Went with Apache Arrow format strings over NumPy format strings
+              because they're more complete from a dataframe perspective
+            - Format strings are mostly useful for datetime specification, and
+              for categoricals.
+            - For categoricals, the format string describes the type of the
+              categorical in the data buffer. In case of a separate encoding of
+              the categorical (e.g. an integer to string mapping), this can
+              be derived from ``self.describe_categorical``.
+            - Data types not included: complex, Arrow-style null, binary, decimal,
+              and nested (list, struct, map, union) dtypes.
+        """
+        dtype = self._col.dtype
+        
+        # Categorical
+        # If it is internal, kind must be categorical (23)
+        # If it is external (call from_dataframe), dtype must give type of the data
+        if self._col.df.is_category(self._col):
+            return (_DtypeKind.CATEGORICAL, 64, 'u', '=') # what should be the default??

```

Comment:
This will have to be revised. See first two items in https://github.com/data-apis/dataframe-api/issues/46#issue-930471826